In [31]:
content = "train"
domain = "sub"
count = 0
sumtoken = 0
maxtoken = 0
mintoken = 100
for line in open("data/token/{}_token_{}.txt".format(content, domain)):
    doctoken = 0
    for sentence in line.split("<sentsep>"):
        senttoken = 0
        for token in sentence.split("<tokesep>"):
            senttoken += 1
        doctoken += senttoken
    
    if doctoken > maxtoken:
        maxtoken = doctoken
#         print(count, maxtoken, line.replace("<sentsep>"," ").replace("<tokesep>", " "))
    if doctoken < mintoken:
        mintoken = doctoken
#         print(count, mintoken, line.replace("<sentsep>"," ").replace("<tokesep>", " "))
    sumtoken += doctoken
    count += 1

In [32]:
print(sumtoken/count, maxtoken, mintoken)

4.552784704904406 24 1


In Subject, max # of token is 24, min # is 1, average # is 4.55

In [33]:
content = "train"
domain = "doc"
count = 0
sumtoken = 0
maxtoken = 0
mintoken = 10000

sumsentence = 0
maxsentence = 0
minsentence = 10000

sumtoknsent = 0
maxtoknsent = 0
mintoknsent = 10000

for line in open("data/token/{}_token_{}.txt".format(content, domain)):
    doctoken = 0
    docsentence = 0
    for sentence in line.split("<sentsep>"):
        senttoken = 0
        for token in sentence.split("<tokesep>"):
            senttoken += 1
        doctoken += senttoken
        
        if senttoken > maxtoknsent:
            maxtoknsent = senttoken
        if senttoken < mintoknsent:
            mintoknsent = senttoken
        sumtoknsent += senttoken

        docsentence += 1
    
    if docsentence > maxsentence:
        maxsentence = docsentence
    if docsentence < minsentence:
        minsentence = docsentence
    
    if doctoken > maxtoken:
        maxtoken = doctoken
    if doctoken < mintoken:
        mintoken = doctoken
    
    sumsentence += docsentence
    sumtoken += doctoken
    count += 1

In [34]:
print(sumtoken/count, maxtoken, mintoken)
print(sumsentence/count, maxsentence, minsentence)
print(sumtoknsent/count, maxtoknsent, mintoknsent)

144.75526461623718 4428 28
7.859310058187864 187 1
144.75526461623718 566 1


In document, average 144.76 tokens, max for 4428, min for 28; average 7.86 sentences, max for 187, min for 1.